rts/Sim/Units/CommandAI/BuilderCaches.h

```diff
@@ -0,0 +1,41 @@
+/* This file is part of the Spring engine (GPL v2 or later), see LICENSE.html */
+
+#ifndef _BUILDER_CACHES_H_
+#define _BUILDER_CACHES_H_
+
+#include "System/UnorderedSet.hpp"
+
+#include <vector>
+
+class CUnit;
+
+class CBuilderCaches
```

Comment:
If it's an instanceless class (not even a singleton needed) perhaps it should be a namespace with regular functions? See https://github.com/beyond-all-reason/spring/pull/1651#discussion_r1725988221